contracts/vault/src/execute.rs

```diff
@@ -0,0 +1,711 @@
+
+    // Calculate each market's utilization and sort by highest utilization
+    let mut utilizations: Vec<(String, Uint128)> = markets
+        .into_iter()
+        .filter_map(|market| {
+            // Query the market status
+            let resp: StatusResp = deps
+                .querier
+                .query(&QueryRequest::Wasm(WasmQuery::Smart {
+                    contract_addr: market.clone(),
+                    msg: to_json_binary(&MarketQueryMsg::Status { price: None })
+                        .expect("Serialize Market Query Msg"),
+                }))
+                .ok()?;
+
+            // Calculate utilization as total_lp + total_xlp, defaulting to 0 on overflow
+            let utilization =
+                (resp.liquidity.total_lp + resp.liquidity.total_xlp).unwrap_or_default();
+
+            // Convert to Uint128
+            let value = Uint128::from(utilization.into_u128().expect("Error LpToken to Uint128"));
+            Some((market, value))
+        })
+        .collect();
+
+    // Sort by utilization, highest to lowest
+    utilizations.sort_by(|a, b| b.1.cmp(&a.1));
+
+    // Sort by utilization, highest to lowest
+    utilizations.sort_by(|a, b| b.1.cmp(&a.1));
+
+    // Sum the total basis points for distribution
+    let total_bps: u16 = config.markets_allocation_bps.iter().sum();
+    if total_bps == 0 {
+        return Err(StdError::generic_err("No allocation percentages defined"));
+    }
+
+    // List of Cosmos messages to execute
+    let mut messages: Vec<CosmosMsg> = vec![];
+    let mut remaining = excess;
+
+    // Distribute excess funds across markets based on basis points
+    for (i, bps) in config.markets_allocation_bps.iter().enumerate() {
+        if let Some((market, _)) = utilizations.get(i) {
+            let amount = excess.multiply_ratio(*bps as u128, total_bps as u128);
+            if !amount.is_zero() {
+                // Create a message to deposit USDC into the market
+                let deposit_msg = WasmMsg::Execute {
+                    contract_addr: market.clone(),
+                    msg: to_json_binary(&MarketExecuteMsg::DepositLiquidity {
+                        stake_to_xlp: false, // Set to true if you want to stake to xLP
+                    })?,
+                    funds: vec![Coin {
+                        denom: config.usdc_denom.clone(),
+                        amount,
+                    }],
+                };
+                messages.push(deposit_msg.into());
+
+                // Update MARKET_ALLOCATIONS with the new allocated amount
+                state::MARKET_ALLOCATIONS.update(
+                    deps.storage,
+                    market.as_str(),
+                    |a| -> Result<Uint128, StdError> { Ok(a.unwrap_or(Uint128::zero()) + amount) },
+                )?;
+
+                // Reduce the remaining amount
+                remaining = remaining.checked_sub(amount).unwrap_or(Uint128::zero());
```

Comment:
Would an overflow here be indicative of a bug somewhere? If so, it's probably better to return an error here. If not, you can use `saturating_sub` instead.